owrx/kiss.py

```diff
@@ -48,11 +49,30 @@ def getConfig(self, port, is_service):
                 lat = "{0:02d}^{1:05.2f}{2}".format(int(lat), (lat - int(lat)) * 60, direction_ns)
                 lon = "{0:03d}^{1:05.2f}{2}".format(int(lon), (lon - int(lon)) * 60, direction_we)
 
+                #Format beacon details
+                symbol  = str(pm["aprs_igate_symbol"])             if "aprs_igate_symbol"  in pm else "R&"
+                power   = str(pm["aprs_igate_power"])              if "aprs_igate_power"   in pm else "0"
+                height  = "HEIGHT=" + str(pm["aprs_igate_height"]) if "aprs_igate_height"  in pm else ""
+                gain    = "GAIN=" + str(pm["aprs_igate_gain"])     if "aprs_igate_gain"    in pm else ""
+                adir    = "DIR=" + str(pm["aprs_igate_dir"])       if "aprs_igate_dir"     in pm else ""
+                freq    = "FREQ=" + str(pm["aprs_igate_freq"])     if "aprs_igate_freq"    in pm else ""
+                tone    = "TONE=" + str(pm["aprs_igate_tone"])     if "aprs_igate_tone"    in pm else ""
+                offset  = "OFFSET=" + str(pm["aprs_igate_offset"]) if "aprs_igate_offset"  in pm else ""
+                comment = str(pm["aprs_igate_comment"])            if "aprs_igate_comment" in pm else "\"OpenWebRX APRS gateway\""
+
+                if((len(comment) > 0) and ((comment[0] != '"') or (comment[len(comment)-1] != '"'))):
+                    comment = "\"" + comment + "\""
+                elif(len(comment) == 0):
+                    comment = "\"\""
+
+                pbeacon= "PBEACON sendto=IG delay=0:30 every=60:00 symbol={symbol} lat={lat} long={lon} POWER={power} {height} {gain} {adir} {freq} {tone} {offset} comment={comment}".format(
+                    symbol=symbol, lat=lat, lon=lon, power=power, height=height, gain=gain, adir=adir, freq=freq, tone=tone, offset=offset, comment=comment )
+
+                logger.info("APRS PBEACON String: " + pbeacon)
+                
                 config += """
-PBEACON sendto=IG delay=0:30 every=60:00 symbol="igate" overlay=R lat={lat} long={lon} comment="OpenWebRX APRS gateway"
-                """.format(
-                    lat=lat, lon=lon
-                )
+                          {pbeacon}
+                          """.format(pbeacon=pbeacon)
```

Comment:
no need to put this through the formatter if there's only a single substitution. You could inline the formatting from above here if you lose the log message.

Reply:
Leftover from experiment I tried.  Will simplify